aws/resource_aws_msk_scram_secret_test.go

```diff
@@ -0,0 +1,359 @@
+    client_subnets  = [aws_subnet.example_subnet_az1.id, aws_subnet.example_subnet_az2.id, aws_subnet.example_subnet_az3.id]
+    ebs_volume_size = 10
+    instance_type   = "kafka.t3.small"
+    security_groups = [aws_security_group.example_sg.id]
+  }
+
+  client_authentication {
+    sasl {
+      scram = true
+    }
+  }
+}
+
+resource "aws_kms_key" "test" {
+  description = "%s-kms-key-msk"
+}
+
+resource "aws_secretsmanager_secret" "test_update" {
+  name       = "AmazonMSK_test_%s_1_update"
+  kms_key_id = aws_kms_key.test.key_id
+}
+
+resource "aws_secretsmanager_secret_version" "test" {
+  secret_id     = aws_secretsmanager_secret.test_update.id
+  secret_string = jsonencode({ username = "%s", password = "foobar" })
+}
+
+resource "aws_secretsmanager_secret" "test2" {
+  name       = "AmazonMSK_test_%s_2_update"
+  kms_key_id = aws_kms_key.test.key_id
+}
+
+resource "aws_secretsmanager_secret_version" "test2" {
+  secret_id     = aws_secretsmanager_secret.test2.id
+  secret_string = jsonencode({ username = "%s", password = "foobar" })
+}
+
+resource "aws_secretsmanager_secret" "test3" {
+  name       = "AmazonMSK_test_%s_3_update"
+  kms_key_id = aws_kms_key.test.key_id
+}
+
+resource "aws_secretsmanager_secret_version" "test3" {
+  secret_id     = aws_secretsmanager_secret.test3.id
+  secret_string = jsonencode({ username = "%s", password = "foobar" })
+}
+
+resource "aws_msk_sasl_scram_secret" "test" {
+  cluster_arn     = aws_msk_cluster.test.arn
+  secret_arn_list = [aws_secretsmanager_secret.test2.arn, aws_secretsmanager_secret.test3.arn]
+}
+
+resource "aws_secretsmanager_secret_policy" "test" {
+  secret_arn = aws_secretsmanager_secret.test_update.arn
+  policy     = <<POLICY
+{
+  "Version" : "2012-10-17",
+  "Statement" : [ {
+    "Sid": "AWSKafkaResourcePolicy",
+    "Effect" : "Allow",
+    "Principal" : {
+      "Service" : "kafka.amazonaws.com"
+    },
+    "Action" : "secretsmanager:getSecretValue",
+    "Resource" : "${aws_secretsmanager_secret.test_update.arn}"
+  } ]
+}
+POLICY
+}
+
+resource "aws_secretsmanager_secret_policy" "test2" {
+  secret_arn = aws_secretsmanager_secret.test2.arn
+  policy     = <<POLICY
+{
+  "Version" : "2012-10-17",
+  "Statement" : [ {
+    "Sid": "AWSKafkaResourcePolicy",
+    "Effect" : "Allow",
+    "Principal" : {
+      "Service" : "kafka.amazonaws.com"
+    },
+    "Action" : "secretsmanager:getSecretValue",
+    "Resource" : "${aws_secretsmanager_secret.test2.arn}"
+  } ]
+}
+POLICY
+}
+
+resource "aws_secretsmanager_secret_policy" "test3" {
+  secret_arn = aws_secretsmanager_secret.test3.arn
+  policy     = <<POLICY
+{
+  "Version" : "2012-10-17",
+  "Statement" : [ {
+    "Sid": "AWSKafkaResourcePolicy",
+    "Effect" : "Allow",
+    "Principal" : {
+      "Service" : "kafka.amazonaws.com"
+    },
+    "Action" : "secretsmanager:getSecretValue",
+    "Resource" : "${aws_secretsmanager_secret.test3.arn}"
+  } ]
+}
+POLICY
+}
+`, rName, rName, rName, rName, rName, rName, rName, rName)
```

Comment:
Similar nit as above -- since we're referencing the same string, let's use the `%[1]` formatter and only pass `rName` here once